columnflow/production/cms/pdf.py

```diff
@@ -108,27 +119,36 @@ def pdf_weights(
             "weight is already included in the LHEWeight.",
         )
 
-    # normalize all weights by the nominal one, assumed to be the first value
-    pdf_weights = events.LHEPdfWeight[~bad_mask, 1:] / pdf_weight_nominal
+    # normalize all 100 weights by the nominal one, assumed to be the first value
+    # (for datasets with 103 weights, the last two weights are related to alpha_s variations)
+    pdf_weights = ak.without_parameters(events.LHEPdfWeight[~bad_mask, 1:101]) / pdf_weight_nominal
```

Comment:
not really part of this PR, but I think that applying this `~bad_mask` does not work as intended, because as soon as some events are masked, the shape of the `pdf_weights` changes and we'll get an error message when trying to add this column to the events (similar issue is also in `scale.py`).
However, I think we want to have the option to raise an exception when there is an invalid number of pdf (scale) weights and maybe provide an option how to handle such events (e.g. setting nominal/up/down to 1)

Reply:
Good point. Let's include this here though.

Reply:
Added.